cpp/src/link_analysis/pagerank.cu

```diff
@@ -164,16 +170,38 @@ int pagerank(IndexType n, IndexType e, IndexType *cscPtr, IndexType *cscInd, Val
 }
 
 //template int pagerank<int, half> (  int n, int e, int *cscPtr, int *cscInd,half *cscVal, half alpha, half *a, bool has_guess, float tolerance, int max_iter, half * &pagerank_vector, half * &residual);
-template int pagerank<int, float> (  int n, int e, int *cscPtr, int *cscInd,float *cscVal, float alpha, float *a, bool has_guess, float tolerance, int max_iter, float * &pagerank_vector, float * &residual);
-template int pagerank<int, double> (  int n, int e, int *cscPtr, int *cscInd,double *cscVal, double alpha, double *a, bool has_guess, float tolerance, int max_iter, double * &pagerank_vector, double * &residual);
+template int pagerank<int, float> (  int n, int e, int *cscPtr, int *cscInd,float *cscVal,
+        int *prsVtx, float *prsVal, int prsLen, bool has_personalization,
+        float alpha, float *a, bool has_guess, float tolerance, int max_iter, float * &pagerank_vector, float * &residual);
+template int pagerank<int, double> (  int n, int e, int *cscPtr, int *cscInd,double *cscVal,
+        int *prsVtx,  double *prsVal, int prsLen, bool has_personalization,
+        double alpha, double *a, bool has_guess, float tolerance, int max_iter, double * &pagerank_vector, double * &residual);
 
 } //namespace cugraph
 
 template <typename WT>
 gdf_error gdf_pagerank_impl (gdf_graph *graph,
-                      gdf_column *pagerank, float alpha = 0.85,
+                      gdf_column *pagerank,
+                      gdf_column *personalization_subset, gdf_column *personalization_values,
+                      float alpha = 0.85,
                       float tolerance = 1e-4, int max_iter = 200,
                       bool has_guess = false) {
+  bool has_personalization = false;
+  int *prsVtx = nullptr;
+  WT  *prsVal = nullptr;
+  int prsLen = 0;
+  GDF_REQUIRE((personalization_subset == nullptr) == (personalization_values == nullptr), GDF_INVALID_API_CALL);
+  if (personalization_subset != nullptr) {
+    has_personalization = true;

```

Comment:
What happens when personalization_subset != nullptr but its size is 0? Does it run the regular PageRank?

Reply:
Good catch. Right now it will still try to run personalized page rank. This needs to be fixed to revert to normal pagerank.

Reply:
Fixed [here](https://github.com/rapidsai/cugraph/pull/300/files?file-filters%5B%5D=.cu&file-filters%5B%5D=.py&owned-by%5B%5D=#diff-c774f1d383f54e52fdc1f235c24c0251R120)